afkak/consumer.py

```diff
@@ -733,6 +733,9 @@ def _handle_fetch_error(self, failure):
             # We've retried until we hit the max delay, log at warn
             log.warning("%r: Still failing fetching messages from kafka: %r",
                         self, failure)
+            if failure.check(OffsetOutOfRangeError):
+                log.info("Reset offset to earliest due to OffsetOutOfRangeError")
+                self._fetch_offset = OFFSET_EARLIEST
```

Comment:
I don't think that this is correct. Do you really only want to check for this error condition once we have retried enough to hit the max delay? And only every other attempt?